packages/mobile/src/identity/privateHashing.ts

```diff
@@ -92,107 +89,42 @@ function* doFetchPhoneHashPrivate(e164Number: string) {
-  const phoneHash = getPhoneHash(e164Number, salt)
-  return {
-    e164Number,
-    phoneHash,
-    salt,
-  }
-}
-
-async function getPhoneNumberSalt(
-  e164Number: string,
-  account: string,
-  contractKit: ContractKit,
-  selfPhoneHash?: string
-) {
-  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Getting phone number salt')
-
+function* getPhoneHashPrivate(e164Number: string, account: string, selfPhoneHash?: string) {
   if (!isE164Number(e164Number)) {
     throw new Error(ErrorMessages.INVALID_PHONE_NUMBER)
   }
 
-  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving blinded message')
-  const base64PhoneNumber = Buffer.from(e164Number).toString('base64')
-  const base64BlindedMessage = (await BlindThresholdBls.blindMessage(base64PhoneNumber)).trim()
-  const base64BlindSig = await postToSignMessage(
-    base64BlindedMessage,
-    account,
-    contractKit,
-    selfPhoneHash
-  )
-  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving unblinded signature')
-  const { pgpnpPubKey } = networkConfig
-  const base64UnblindedSig = await BlindThresholdBls.unblindMessage(base64BlindSig, pgpnpPubKey)
-  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Converting sig to salt')
-  return getSaltFromThresholdSignature(base64UnblindedSig)
-}
-
-interface SignMessageRequest {
-  account: string
-  blindedQueryPhoneNumber: string
-  hashedPhoneNumber?: string
-}
-
-interface SignMessageResponse {
-  success: boolean
-  combinedSignature: string
-}
-
-// Send the blinded message off to the phone number privacy service and
-// get back the theshold signed blinded message
-async function postToSignMessage(
-  base64BlindedMessage: string,
-  account: string,
-  contractKit: ContractKit,
-  selfPhoneHash?: string
-) {
-  const body: SignMessageRequest = {
-    account,
-    blindedQueryPhoneNumber: base64BlindedMessage,
-    hashedPhoneNumber: selfPhoneHash,
+  const authSigner = yield call(getAuthSignerForAccount, account)
+  // Unlock the account if the authentication is signed by the wallet
+  if (authSigner.authenticationMethod === AuthenticationMethod.WALLETKEY) {
+    const success: boolean = yield call(unlockAccount, account)
+    if (!success) {
+      throw new Error(ErrorMessages.INCORRECT_PIN)
+    }
   }
 
+  const { pgpnpPubKey } = networkConfig
+  const blsBlindingClient = new ReactBlsBlindingClient(pgpnpPubKey)
+
   try {
-    const response = await postToPhoneNumPrivacyService<SignMessageResponse>(
+    return yield call(
+      getPhoneNumberIdentifier,
+      e164Number,
       account,
-      contractKit,
-      body,
-      SIGN_MESSAGE_ENDPOINT
+      authSigner,
+      networkConfig,
+      selfPhoneHash,
+      DeviceInfo.getVersion(),
+      blsBlindingClient
     )
-    return response.combinedSignature
   } catch (error) {
     if (error.message === ErrorMessages.PGPNP_QUOTA_ERROR) {
-      throw new Error(ErrorMessages.SALT_QUOTA_EXCEEDED)
+      throw new Error(ErrorMessages.MATCHMAKING_QUOTA_EXCEEDED)
```

Comment:
This file is for getting salts, not matchmaking. Was this change intentional?

Reply:
Good eye, thank you